hamilton/src/hybrid_planner.cpp

```diff
@@ -0,0 +1,576 @@
+		y_axes.points.reserve(2*poses.size());
+		x_axes.points.reserve(2*poses.size());
+		line.points.reserve(poses.size());
+		geometry_msgs::Point p_start,p_end;
+		double distance = 0;
+		Eigen::Affine3d prev = poses[0];
+		for(unsigned int i = 0; i < poses.size(); i++)
+		{
+		const Eigen::Affine3d& pose = poses[i];
+		distance = (pose.translation() - prev.translation()).norm();
+
+		tf::pointEigenToMsg(pose.translation(),p_start);
+
+		if(distance > config_.min_point_distance)
+		{
+		  Eigen::Affine3d moved_along_x = pose * Eigen::Translation3d(AXIS_LINE_LENGHT,0,0);
+		  tf::pointEigenToMsg(moved_along_x.translation(),p_end);
+		  x_axes.points.push_back(p_start);
+		  x_axes.points.push_back(p_end);
+
+		  Eigen::Affine3d moved_along_y = pose * Eigen::Translation3d(0,AXIS_LINE_LENGHT,0);
+		  tf::pointEigenToMsg(moved_along_y.translation(),p_end);
+		  y_axes.points.push_back(p_start);
+		  y_axes.points.push_back(p_end);
+
+		  Eigen::Affine3d moved_along_z = pose * Eigen::Translation3d(0,0,AXIS_LINE_LENGHT);
+		  tf::pointEigenToMsg(moved_along_z.translation(),p_end);
+		  z_axes.points.push_back(p_start);
+		  z_axes.points.push_back(p_end);
+
+		  // saving previous
+		  prev = pose;
+		}
+
+		line.points.push_back(p_start);
+		}
+
+		markers_msg.markers.push_back(x_axes);
+		markers_msg.markers.push_back(y_axes);
+		markers_msg.markers.push_back(z_axes);
+		markers_msg.markers.push_back(line);
+
+		marker_publisher_.publish(markers_msg);
+
+	}
+	// // HybridPlanner::appendJointSpacePoint(std::std::vector<double> joints)
+	// // {
+	// //   	group.setJointValueTarget(joints);
+	// //   	// group.move()
+	// // }
+
+	bool HybridPlanner::appendFreeSpaceSegment(std::vector<geometry_msgs::Pose> waypoints)
+	{
+		// moveit_trajectory_.append(waypoint);
+		// moveit_trajectory_.push_back(waypoints);
+
+		// TODO TOCHECK maintain waypoints and call computecartesianpath in the plan / execute function
+		// We want the cartesian path to be interpolated at a resolution of 1 cm which is why we will specify 0.01 as the max step in cartesian translation. 
+		// We will specify the jump threshold as 0.0, effectively disabling it.
+
+		double fraction = moveit_group_ptr_-> computeCartesianPath(waypoints,
+		                                             config_.step_size,  // eef_step
+		                                             config_.jump_thresh,   // jump_threshold
+		                                             moveit_robot_traj_);	
+
+		robot_trajectory::RobotTrajectory robot_traj_segment(kinematic_model_, config_.group_name);
+		robot_traj_segment.setRobotTrajectoryMsg(	*kinematic_state_, moveit_robot_traj_);
+		// robot_state::RobotState last_robot_state = robot_traj_segment.getLastWayPoint();
+
+		overall_robot_traj_.setRobotTrajectoryMsg(*kinematic_state_, moveit_robot_traj_);
+
+		overall_robot_traj_.append(robot_traj_segment,
+    		robot_traj_segment.getWaypointDurationFromStart(robot_traj_segment.getWayPointCount()));
+
```

Comment:
This function is incomplete. 
So by now, after calling appendFreeSpaceSegment() and appendDescartesCartPointSegment() multiple times, followed by a planProcessPath, we have the freeSpaceTrajectories in robot_trajectory::RobotTrajectory overall_robot_traj_ and the processTrajectories in moveit_robot_traj moveit_robot_traj_
Note : typedef moveit_msgs::RobotTrajectory moveit_robot_traj
Then I need to stitch them together (somehow), and I just have a feeling there is time messed up somewhere in the points or this isn't the best approach.